client/transaction-pool/src/tests.rs

```diff
@@ -14,42 +14,70 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-
 use super::*;
 
+use crate::{BasicPool, MaintainedTransactionPool};
 use codec::Encode;
 use futures::executor::block_on;
-use sc_transaction_graph::{self, Pool};
-use substrate_test_runtime_client::{runtime::{AccountId, Block, Hash, Index, Extrinsic, Transfer}, AccountKeyring::{self, *}};
+use parking_lot::RwLock;
+use sc_transaction_graph::{self, ExHash, Pool};
 use sp_runtime::{
 	generic::{self, BlockId},
-	traits::{Hash as HashT, BlakeTwo256},
-	transaction_validity::{TransactionValidity, ValidTransaction},
+	traits::{BlakeTwo256, Hash as HashT},
+	transaction_validity::{TransactionValidity, ValidTransaction, TransactionValidityError, InvalidTransaction},
+};
+use std::collections::HashSet;
+use substrate_test_runtime_client::{
+	runtime::{AccountId, Block, BlockNumber, Extrinsic, Hash, Header, Index, Transfer},
+	AccountKeyring::{self, *},
 };
 
 struct TestApi {
-	pub modifier: Box<dyn Fn(&mut ValidTransaction) + Send + Sync>,
+	pub modifier: RwLock<Box<dyn Fn(&mut ValidTransaction) + Send + Sync>>,
```

Comment:
Could we integrate this structure under single `RwLock`? It's test code and I think it would benefit more from simplicity than performance, and better if no one deadlocks their tests by messing up locking order.

Reply:
I was planning to refactor this TestApi completely in the next PR, can we do it then?

Reply:
Yes, please maybe make an issue for that to not forget.